Hw2.1

Julia-Berezhnova opened the pull request:
Homework 2.1 is added

Julia-Berezhnova committed:
Homework 2.1 is added

Julia-Berezhnova committed:
Refactoring is done, tests are added

yurii-litvinov commented on hw2_1/hw2_1/ArrayStack.cs:
```diff
@@ -0,0 +1,81 @@
+			else 
+			{
+				return false;
+			}	
```
это можно вообще в одну строчку записать:
public bool isEmpty() => this.currentSize == 0;

yurii-litvinov commented on hw2_1/hw2_1/ArrayStack.cs:
```diff
@@ -0,0 +1,81 @@
+		/// </summary>
+		public int Pop()
+		{
+			--currentSize;
```
тут бы неплохо проверку, потому что если кто-то не проверит стек на пустоту, вообще упадёт всё

yurii-litvinov commented on hw2_1/hw2_1/hw2_1Test.cs:
```diff
@@ -0,0 +1,30 @@
+		{
+			Assert.AreEqual (true, myStack.isEmpty ());
+		}
+	}
```
Такому набору тестов удовлетворит даже стек, который просто хранит в себе одно последнее значение. Кроме того, тесты должны быть независимы, то есть если один из тестов вообще даже не запускали, остальные должны проходить. Так что myStack имеет смысл создавать прямо внутри теста или воспользоваться http://www.nunit.org/index.php?p=setup&r=2.2.10 (что правильнее)

Julia-Berezhnova committed:
Homework 2.1 is fixed

yurii-litvinov commented on hw2_1/hw2_1/hw2_1Test.cs:
```diff
@@ -0,0 +1,60 @@
+		}
+
+		[Test ()]
+		public void Push()
```
Обычно пишут что-нибудь вроде PushTest

yurii-litvinov commented on hw2_1/hw2_1/hw2_1Test.cs:
```diff
@@ -0,0 +1,60 @@
+	[TestFixture ()]
+	public class hw2_1Test
+	{
+		[SetUp()] public void InitialState()
```
атрибуты лучше писать над методом

Julia-Berezhnova committed:
Refactoring is done

Julia-Berezhnova committed:
Stack.Pop() is modified

yurii-litvinov commented:
говорит, This branch has conflicts that must be resolved. поправите?

Julia-Berezhnova commented:
Я так понимаю, надо так:
$ git checkout hw2.1
$ git fetch origin
$ git merge origin/master
?

2016-03-12 15:25 GMT+03:00 Yurii Litvinov notifications@github.com:

> говорит, This branch has conflicts that must be resolved. поправите?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Julia-Berezhnova/Semester-2/pull/6#issuecomment-195730034
> .

yurii-litvinov commented:
Ага. Или сразу 

```
 git checkout hw2.1
 git pull origin master
```

pull --- это fetch (только одной ветки, правда) + merge

Julia-Berezhnova commented:
сделала, как Вы сказали. пишет automatic merge failed; fix conflicts and
them commit the result. в папке появились замердженные папки hw1.1..4 +
hw2.1. делать git add?

2016-03-12 21:48 GMT+03:00 Yurii Litvinov notifications@github.com:

> Ага. Или сразу
> 
>  git checkout hw2.1
>  git pull origin master
> 
> pull --- это fetch (только одной ветки, правда) + merge
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Julia-Berezhnova/Semester-2/pull/6#issuecomment-195784504
> .

yurii-litvinov commented:
а оно должно было по идее написать, где там конфликты

Julia-Berezhnova commented:
julia@home:~/Homework$ git pull origin master
remote: Counting objects: 9, done.
remote: Compressing objects: 100% (5/5), done.
remote: Total 9 (delta 3), reused 0 (delta 0), pack-reused 4
Unpacking objects: 100% (9/9), done.
From https://github.com/Julia-Berezhnova/Semester-2
- branch            master     -> FETCH_HEAD
  0871f29..27c254d  master     -> origin/master
  CONFLICT (modify/delete): hw1_4/hw1_4/Program.cs deleted in HEAD and
  modified in 27c254dd83996708d0942ce9851e0aa9f0269fc3. Version
  27c254dd83996708d0942ce9851e0aa9f0269fc3 of hw1_4/hw1_4/Program.cs left in
  tree.
  Automatic merge failed; fix conflicts and then commit the result.

2016-03-12 22:51 GMT+03:00 Yurii Litvinov notifications@github.com:

> а оно должно было по идее написать, где там конфликты
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Julia-Berezhnova/Semester-2/pull/6#issuecomment-195797949
> .

yurii-litvinov commented:
ага, пишет, что hw1_4/hw1_4/Program.cs в мастере модифицирована, а тут удалена. можно на неё сделать git add и будет ок, наверное

Julia-Berezhnova commented:
а там в папке нету hw1.4

12 марта 2016 г., 23:00 пользователь Yurii Litvinov <
notifications@github.com> написал:

> ага, пишет, что hw1_4/hw1_4/Program.cs в мастере модифицирована, а тут
> удалена. можно на неё сделать git add и будет ок, наверное
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Julia-Berezhnova/Semester-2/pull/6#issuecomment-195799554
> .

yurii-litvinov commented:
а что говорит git status?

Julia-Berezhnova commented:
On branch hw2.1
Your branch is up-to-date with 'origin/hw2.1'.
You have unmerged paths.
  (fix conflicts and run "git commit")

Changes to be committed:

```
new file:   hw1/hw1/hw1.sln
new file:   hw1/hw1/hw1/Program.cs
new file:   hw1/hw1/hw1/Properties/AssemblyInfo.cs
new file:   hw1/hw1/hw1/hw1_1.csproj
new file:   hw1_3/hw1_3.sln
new file:   hw1_3/hw1_3/Program.cs
new file:   hw1_3/hw1_3/Properties/AssemblyInfo.cs
new file:   hw1_3/hw1_3/hw1_3.csproj
new file:   hw1_5/hw1_5.sln
new file:   hw1_5/hw1_5/Program.cs
new file:   hw1_5/hw1_5/Properties/AssemblyInfo.cs
new file:   hw1_5/hw1_5/hw1_5.csproj
new file:   hw2/hw1_2/hw1_2.sln
new file:   hw2/hw1_2/hw1_2/Program.cs
new file:   hw2/hw1_2/hw1_2/Properties/AssemblyInfo.cs
new file:   hw2/hw1_2/hw1_2/hw1_2.csproj
```

Unmerged paths:
  (use "git add/rm <file>..." as appropriate to mark resolution)

```
deleted by us:   hw1_4/hw1_4/Program.cs
```

Untracked files:
  (use "git add <file>..." to include in what will be committed)

```
FETCH_HEAD
```

julia@home:~/Homework$ git status
On branch hw2.1
Your branch is up-to-date with 'origin/hw2.1'.
You have unmerged paths.
  (fix conflicts and run "git commit")

Changes to be committed:

```
new file:   hw1/hw1/hw1.sln
new file:   hw1/hw1/hw1/Program.cs
new file:   hw1/hw1/hw1/Properties/AssemblyInfo.cs
new file:   hw1/hw1/hw1/hw1_1.csproj
new file:   hw1_3/hw1_3.sln
new file:   hw1_3/hw1_3/Program.cs
new file:   hw1_3/hw1_3/Properties/AssemblyInfo.cs
new file:   hw1_3/hw1_3/hw1_3.csproj
new file:   hw1_5/hw1_5.sln
new file:   hw1_5/hw1_5/Program.cs
new file:   hw1_5/hw1_5/Properties/AssemblyInfo.cs
new file:   hw1_5/hw1_5/hw1_5.csproj
new file:   hw2/hw1_2/hw1_2.sln
new file:   hw2/hw1_2/hw1_2/Program.cs
new file:   hw2/hw1_2/hw1_2/Properties/AssemblyInfo.cs
new file:   hw2/hw1_2/hw1_2/hw1_2.csproj
```

Unmerged paths:
  (use "git add/rm <file>..." as appropriate to mark resolution)

```
deleted by us:   hw1_4/hw1_4/Program.cs
```

Untracked files:
  (use "git add <file>..." to include in what will be committed)

```
FETCH_HEAD
```

yurii-litvinov commented:
а если всё равно сделать git add hw1_4/hw1_4/Program.cs?

Julia-Berezhnova commented:
julia@home:~/Homework$ git add hw1_4/hw1_4/Program.cs
julia@home:~/Homework$ git status
On branch hw2.1
Your branch is up-to-date with 'origin/hw2.1'.
All conflicts fixed but you are still merging.
  (use "git commit" to conclude merge)

Changes to be committed:

```
new file:   hw1/hw1/hw1.sln
new file:   hw1/hw1/hw1/Program.cs
new file:   hw1/hw1/hw1/Properties/AssemblyInfo.cs
new file:   hw1/hw1/hw1/hw1_1.csproj
new file:   hw1_3/hw1_3.sln
new file:   hw1_3/hw1_3/Program.cs
new file:   hw1_3/hw1_3/Properties/AssemblyInfo.cs
new file:   hw1_3/hw1_3/hw1_3.csproj
new file:   hw1_4/hw1_4/Program.cs
new file:   hw1_5/hw1_5.sln
new file:   hw1_5/hw1_5/Program.cs
new file:   hw1_5/hw1_5/Properties/AssemblyInfo.cs
new file:   hw1_5/hw1_5/hw1_5.csproj
new file:   hw2/hw1_2/hw1_2.sln
new file:   hw2/hw1_2/hw1_2/Program.cs
new file:   hw2/hw1_2/hw1_2/Properties/AssemblyInfo.cs
new file:   hw2/hw1_2/hw1_2/hw1_2.csproj
```

Untracked files:
  (use "git add <file>..." to include in what will be committed)

```
FETCH_HEAD
```

2016-03-12 23:35 GMT+03:00 Yurii Litvinov notifications@github.com:

> а если всё равно сделать git add hw1_4/hw1_4/Program.cs?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Julia-Berezhnova/Semester-2/pull/6#issuecomment-195805324
> .

yurii-litvinov commented:
теперь commit и push?

Julia-Berezhnova committed:
trial to fix the conflict

Julia-Berezhnova commented:
julia@home:~/Homework$ git commit -a -m "trial to fix the conflict"
[hw2.1 ea0ee52] trial to fix the conflict
julia@home:~/Homework$ git push -u origin master
To https://github.com/Julia-Berezhnova/Semester-2.git
 ! [rejected]        master -> master (non-fast-forward)
error: failed to push some refs to '
https://github.com/Julia-Berezhnova/Semester-2.git'
hint: Updates were rejected because a pushed branch tip is behind its remote
hint: counterpart. Check out this branch and integrate the remote changes
hint: (e.g. 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.

2016-03-13 14:39 GMT+03:00 Yurii Litvinov notifications@github.com:

> теперь commit и push?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Julia-Berezhnova/Semester-2/pull/6#issuecomment-195940085
> .

yurii-litvinov commented:
не git push -u origin master, просто git push -u, надо ведь запушить в ветку hw2.1, а не мастер

Julia-Berezhnova commented:
Теперь ок?
julia@home:~/Homework$ git push -u
Counting objects: 4, done.
Delta compression using up to 8 threads.
Compressing objects: 100% (2/2), done.
Writing objects: 100% (4/4), 412 bytes | 0 bytes/s, done.
Total 4 (delta 1), reused 0 (delta 0)
To https://github.com/Julia-Berezhnova/Semester-2.git
   742c1f7..ea0ee52  hw2.1 -> hw2.1
Branch hw2.1 set up to track remote branch hw2.1 from origin.

2016-03-13 15:41 GMT+03:00 Yurii Litvinov notifications@github.com:

> не git push -u origin master, просто git push -u, надо ведь запушить в
> ветку hw2.1, а не мастер
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Julia-Berezhnova/Semester-2/pull/6#issuecomment-195949517
> .

yurii-litvinov commented:
да, конфликт разрешился :) но в диффе всё равно есть какие-то файлы из hw1.4, так что если этот пуллреквест замерджить, половина первой домашки погибнет. можете восстановить не по делу удалённые в этой ветке файлы?

Julia-Berezhnova commented:
Как?

13 марта 2016 г., 15:47 пользователь Yurii Litvinov <
notifications@github.com> написал:

> да, конфликт разрешился :) но в диффе всё равно есть какие-то файлы из
> hw1.4, так что если этот пуллреквест замерджить, половина первой домашки
> погибнет. можете восстановить не по делу удалённые в этой ветке файлы?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Julia-Berezhnova/Semester-2/pull/6#issuecomment-195950818
> .

yurii-litvinov commented:
http://stackoverflow.com/questions/953481/find-and-restore-a-deleted-file-in-a-git-repository

Julia-Berezhnova committed:
fixing the promlems

Julia-Berezhnova committed:
fixing the problems

Julia-Berezhnova committed:
fixing the problems

Julia-Berezhnova committed:
Hopefully, he problem with white spaces is solved

yurii-litvinov commented on hw2_1/hw2_1/ArrayStack.cs:
```diff
@@ -0,0 +1,82 @@
+            }
+            else
+            {
+                return false;
```
тут бы тоже бросать исключение, раз в Pop Вы его бросаете
